app/services/EssttpBarsService.scala

```diff
@@ -42,14 +49,30 @@ class EssttpBarsService @Inject() (barsService: BarsService, auditService: Audit
   )(implicit requestHeader: RequestHeader): Future[Either[BarsError, VerifyResponse]] = {
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(requestHeader)
 
+    val taxId = journey match {
+      case account: Stages.ChosenTypeOfBankAccount     => account.taxId
```

Comment:
you could shorten this by using `BeforeComputeTaxId` and `AfterComputeTaxId` like this perhaps:
```scala
journey match {
  case j: AfterComputeTaxId => j.taxId
  case _: BeforeComputeTaxId => throw SomeError("expected tax id but blah")
}
```

Reply:
I'll try that